shanoir-ng-datasets/src/main/java/org/shanoir/ng/configuration/amqp/RabbitMQDatasetsService.java

```diff
@@ -89,41 +95,43 @@ public void receiveMessage(String commandArrStr) {
 	@RabbitListener(queues = RabbitMQConfiguration.STUDY_NAME_UPDATE_QUEUE)
 	@RabbitHandler
 	public void receiveStudyNameUpdate(final String studyStr) {
-		ObjectMapper objectMapper = new ObjectMapper();
-		IdName receivedStudy = new IdName();
-		try {
-			receivedStudy = objectMapper.readValue(studyStr, IdName.class);
-			Study existingStudy = studyRepository.findOne(receivedStudy.getId());
-			if (existingStudy != null) {
-				// update existing study's name
-				existingStudy.setName(receivedStudy.getName());
-				studyRepository.save(existingStudy);
-			} else {
-				// create new study
-				Study newStudy = new Study(receivedStudy.getId(), receivedStudy.getName());
-				studyRepository.save(newStudy);
-			}
-		} catch (IOException e) {
-			LOG.error("Could not read value transmit as Study class through RabbitMQ");
-			throw new AmqpRejectAndDontRequeueException("Something went wrong deserializing the event." + e.getMessage());
-		}
+		receiveAndUpdateIdNameEntity(studyStr, Study.class, studyRepository);
 	}
 
 	@Transactional
 	@RabbitListener(queues = RabbitMQConfiguration.SUBJECT_NAME_UPDATE_QUEUE)
 	@RabbitHandler
 	public void receiveSubjectNameUpdate(final String subjectStr) {
+		receiveAndUpdateIdNameEntity(subjectStr, Subject.class, subjectRepository);
+	}
+	
+	@Transactional
+	@RabbitListener(queues = RabbitMQConfiguration.CENTER_NAME_UPDATE_QUEUE)
+	@RabbitHandler
+	public void receiveCenterNameUpdate(final String centerStr) {
+		receiveAndUpdateIdNameEntity(centerStr, Center.class, centerRepository);
+	}
+	
+	private <T extends IdName> void receiveAndUpdateIdNameEntity(final String receivedStr, final Class<T> clazz, final CrudRepository<T, Long> repository) {
 		ObjectMapper objectMapper = new ObjectMapper();
-		IdName receivedSubject = new IdName();
+		IdName received = new IdName();
 		try {
-			receivedSubject = objectMapper.readValue(subjectStr, IdName.class);
-			Subject existingSubject = subjectRepository.findOne(receivedSubject.getId());
-			if (existingSubject != null) {
-				existingSubject.setName(receivedSubject.getName());
-				subjectRepository.save(existingSubject);
+			received = objectMapper.readValue(receivedStr, IdName.class);
+			T existing = repository.findOne(received.getId());
+			if (existing != null) {
+				// update existing entity's name
+				existing.setName(received.getName());
+				repository.save(existing);
 			} else {
-				Subject newSubject = new Subject(receivedSubject.getId(), receivedSubject.getName());
-				subjectRepository.save(newSubject);
+				// create new entity
+				try {
+					T newOne = clazz.newInstance();
+					newOne.setId(received.getId());
+					newOne.setName(received.getName());
+					repository.save(newOne);
+				} catch ( SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException e) {
+					throw new IllegalStateException("Cannot instanciate " + clazz.getSimpleName() + " class through reflection. It is a programming error.", e);
```

Comment:
This is quite "dangerous" to use IllegalStateException in a @RabbitHandlerMethod
=> If the catched exception is not AmqpRejectAndDontRequeueException, then the message well be requeued and then it'll loop over and over

Reply:
@Nunien Can you update this to use something else than an IllegalStateException so that it does not requeue in an infinite loop ?

Reply:
Yes !